packages/headless/src/features/commerce/search/search-actions.ts

```diff
@@ -75,13 +97,14 @@ export const executeSearch = createAsyncThunk<
 
     return {
       response: fetched.success,
-      queryExecuted: state.commerceQuery?.query,
+      originalQuery: query,
+      queryExecuted: querySelector(state),
```

Comment:
`queryExecuted` needs to be reconciled with what's returned by the API (ie: if there's correction, we use that value, otherwise, yes, selecting the value from the state is okay.

At this point in the execution, the value in the state did not have a chance yet to be modified by the did-you-mean-slice (which does it's job when this `executeSearch` is `fulfilled`.

Reply:
Ok, thank you, that makes sense! I think I've got the right fix in https://github.com/coveo/ui-kit/pull/3906/commits/45aba476ce43e8a6b773126ee67546b033e26f06